dist/tools/bmp/bmp.py

```diff
@@ -0,0 +1,274 @@
+    if args.action == 'term':
+        port = u[0].device
+        if args.port:
+            port = args.port
+        elif args.serial:
+            port = search_serial(args.serial, u)
+            assert port, "no BMP with this serial found"
+        os.system(args.term_cmd % port)
+        sys.exit(0)
+    else:
+        print("found following Black Magic GDB servers:")
+        for i, s in enumerate(g):
+            print("\t[%s]" % s.device, end=' ')
+            if len(s.serial_number) > 1:
+                print("Serial:", s.serial_number, end=' ')
+            if i == 0:
+                print("<- default", end=' ')
+            print('')
+
+        port = g[0].device
+        if args.port:
+            port = args.port
+        elif args.serial:
+            port = search_serial(args.serial, g)
+            assert port, "no BMP with this serial found"
+
+        print('connecting to [%s]...' % port)
+
+        args.file = args.file if args.file else ''
+        args.gdb_path = find_suitable_gdb(args.gdb_path)
+        if not args.gdb_path:
+            print("CANNOT LOCATE SUITABLE GDB EXECUTABLE!", file=sys.stderr)
+            sys.exit(-1)
+
+        # debug mode, opens GDB shell with options
+        if args.action == 'debug':
+            gdb_args = ['-ex \'target extended-remote %s\'' % port]
+            if args.tpwr:
+                gdb_args.append('-ex \'monitor tpwr enable\'')
+            if args.connect_srst:
+                gdb_args.append('-ex \'monitor connect_srst enable\'')
+            if args.jtag:
+                gdb_args.append('-ex \'monitor jtag_scan\'')
+            else:
+                gdb_args.append('-ex \'monitor swdp_scan\'')
+            gdb_args.append('-ex \'attach %s\'' % args.attach)
+            os.system(" ".join(['\"' + args.gdb_path + '\"'] + gdb_args + [args.file]))
+            sys.exit(0)
+
+        # open GDB in machine interface mode
+        gdbmi = GdbController(gdb_path=args.gdb_path, gdb_args=["--nx", "--quiet", "--interpreter=mi2", args.file])
+        assert gdb_write_and_wait_for_result(gdbmi, '-target-select extended-remote %s' % port, 'connecting',
+                                             expected_result='connected')
+        # set options
+        if args.connect_srst:
+            gdbmi.write('monitor connect_srst enable', timeout_sec=TIMEOUT)
+        if args.tpwr:
+            gdbmi.write('monitor tpwr enable', timeout_sec=TIMEOUT)
+
+        # scan for targets
+        if not args.jtag:
+            print("scanning using SWD...")
+            res = gdbmi.write('monitor swdp_scan', timeout_sec=TIMEOUT)
+        else:
+            print("scanning using JTAG...")
+            res = gdbmi.write('monitor jtag_scan', timeout_sec=TIMEOUT)
+        targets = detect_targets(gdbmi, res)
+        assert len(targets) > 0, "no targets found"
+        print("found following targets:")
+        for t in targets:
+            print("\t%s" % t)
+        print("")
```

Comment:
How about a separate function for this, e.g. `connect_to_bmp()`?

Reply:
this part became `connect_to_target`